lib/block/block.go

```diff
@@ -74,14 +73,15 @@ func getBlockKeyPrefixHeight(height uint64) string {
 func (b Block) NewBlockKeyConfirmed() string {
 	return fmt.Sprintf(
 		"%s%s-%s%s",
-		common.BlockPrefixConfirmed, b.Confirmed,
+		common.BlockPrefixConfirmed, b.ProposedTime,
 		common.EncodeUint64ToByteSlice(b.Height),
 		common.GetUniqueIDFromUUID(),
 	)
 }
 
 func (b *Block) Save(st *storage.LevelDBBackend) (err error) {
 	key := getBlockKey(b.Hash)
+	b.Confirmed = common.NowISO8601()
```

Comment:
Are we sure the `Save` time is deterministic ?

Reply:
It is not deterministic. And not need to be deterministic. Because it is not used for calculating Hash.